### DISCUSSION TITLE
Fields created by `RichEditor` are shown with HTML tags in table

### DISCUSSION THREAD

**Discussion by mloru** [filamentphp/filament#3080] [Help] [answered]
**Package**
filament/filament

**Package Version**
v2.13.14

**Laravel Version**
v9.18.0

**Livewire Version**
v2.10.6

**PHP Version**
v8.1.5

**Bug Description**
I'm new to Filament, I'm trying to create a simple resource with just a couple of fields.
One of them is managed by `RichEditor` in the `form()` method, but once a record is created, the table view shows this field including the tags. Is this a known/intended behavior? If so, how can I customize the field before it is loaded in the table?

**To reproduce**

1. Create a resource that includes a `Forms\Components\RichEditor` field:

```php
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\DatePicker::make('date')
                                           ->required(),

                Forms\Components\RichEditor::make('text')
                                           ->required(),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('date')
                                         ->date(),
                Tables\Columns\TextColumn::make('text'),
            ])
            ->filters([
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\DeleteBulkAction::make(),
            ]);
    }
```

2. Create a record for that resource using the UI, including some rich text such as bold or italic text
3. Open the table view of that resource. The text will be shown as in the screenshot below

![image](https://user-images.githubusercontent.com/1322773/178141261-b3b12cd5-7933-4e2d-862d-c5748b017117.png)

**Comment by ryangjchandler** [comment]
This is expected since the HTML stored by the field is escaped when echo'd out. If we didn't escape it, it would be vulnerable to XSS. 

You can customise the output using `formatStateUsing()`. If you want the HTML to be output correctly, you can return an `HtmlString`. 

If you want to remove the tags, run the `$state` through `strip_tags()`.

**Comment by danharrin** [comment, marked as answer]
I think we have an `html()` helper for the column

**Reply by yablochnyi** [reply]
Is it possible to shorten the field? since the text is too long, the table is large, you need to display up to 50 characters

**Reply by danharrin** [reply]
Truncating HTML content is not trivial

**Comment by mloru** [comment]
I ended up using `html()`, thanks to both of you :)
